packages/core/src/utils.ts

```diff
@@ -975,24 +995,47 @@ export const toOpenZeppelinProxyType = (
 }
 
 export const getOpenZeppelinValidationOpts = (
-  proxyType: ProxyType
+  proxyType: ProxyType,
+  contractConfig: UserContractConfig
 ): Required<ValidationOptions> => {
-  return {
+  type UnsafeAllow = Required<ValidationOptions>['unsafeAllow']
+
+  const unsafeAllow: UnsafeAllow = [
+    'state-variable-assignment',
+    'constructor',
+    'state-variable-immutable',
+    'missing-public-upgradeto',
+  ]
+  if (contractConfig.unsafeAllow?.delegatecall) {
+    unsafeAllow.push('delegatecall')
+  }
+  if (contractConfig.unsafeAllow?.selfdestruct) {
+    unsafeAllow.push('selfdestruct')
+  }
+
+  const options = {
     kind: toOpenZeppelinProxyType(proxyType),
-    unsafeAllow: [],
+    unsafeAllow,
     unsafeAllowCustomTypes: false,
-    unsafeAllowLinkedLibraries: false,
-    unsafeAllowRenames: false,
-    unsafeSkipStorageCheck: false,
+    unsafeAllowRenames: contractConfig.unsafeAllowRenames ?? false,
+    unsafeSkipStorageCheck: contractConfig.unsafeSkipStorageCheck ?? false,
   }
+
+  return withValidationDefaults(options)
 }
 
-export const getOpenZeppelinStorageLayout = (
+export const getOpenZeppelinUpgradableContract = (
```

Comment:
Since open zeppelin automatically runs the safety checks when you create a `UpgradableContract`, I decided to refactor this function so that it returns the entire upgradable contract so it could be used for both the storage layout check and the rest of the safety checks.